frontend/src/pages/document/[docId].tsx

```diff
@@ -1,80 +1,83 @@
+import TabbedNav from "@components/nav/TabbedNav";
+import { ExternalLinkIcon } from "@components/svg/Icons";
+import { convertDate } from "@utils/timedate";
+import { initialSummaryLength } from "@constants/document";
+import { truncateString } from "../../helpers";
+
+type TEvent = {
+  name: string;
+  created_ts: string;
+  date: string;
+  description: string;
+};
 
 const DocumentCoverPage = () => {
   const [showFullSummary, setShowFullSummary] = useState(false);
-  const [summary, setSummary] = useState('');
-  const { t, i18n, ready } = useTranslation('searchStart');
+  const [summary, setSummary] = useState("");
   const router = useRouter();
   const structureData = useSortAndStructure();
-  const collapsedLength = 1400;
 
   const documentQuery = useDocumentDetail(router.query.docId as string);
-
-  const { isFetching, isError, error, data } = documentQuery;
+  const { isFetching } = documentQuery;
   const { data: { data: page } = {} } = documentQuery;
 
-  const [year, day, month] = convertDate(page?.publication_ts);
+  const [year] = convertDate(page?.publication_ts);
+
+  useEffect(() => {
+    if (page?.description) {
+      toggleSummary();
+    }
+  }, [page, showFullSummary]);
+
+  useEffect(() => {
+    if (router?.query.docId) {
+      documentQuery.refetch();
+    }
+  }, [router.query]);
 
   const toggleSummary = () => {
     const text = page?.description;
     if (showFullSummary) {
       setSummary(text);
     } else {
-      setSummary(truncateString(text, collapsedLength));
+      setSummary(truncateString(text, initialSummaryLength));
     }
   };
 
   const renderSourceLink = () => {
     let link: string;
-    if (page.content_type === 'application/pdf' && page.url.length) {
+    if (page.content_type === "application/pdf" && page.url.length) {
       link = page.url;
     } else if (page.source_url.length) {
       link = page.source_url;
     }
 
     if (!link) return null;
+
     return (
       <p className="mt-4">
-        <a
-          href={link}
-          target="_blank"
-          rel="noopener noreferrer nofollow"
-          className="text-blue-500 underline flex items-center font-medium hover:text-indigo-600 transition duration-300"
-        >
+        <a href={link} target="_blank" rel="noopener noreferrer nofollow" className="text-blue-500 underline font-medium hover:text-indigo-600 transition duration-300">
           <span className="mr-1">Link to source document</span>
-
-          <ExternalLinkIcon height="16" width="16" />
+          <span className="inline-block">
+            <ExternalLinkIcon height="16" width="16" />
+          </span>
         </a>
       </p>
     );
   };
 
-  useEffect(() => {
-    if (page?.description) {
-      toggleSummary();
-    }
-  }, [page, showFullSummary]);
-
-  useEffect(() => {
-    if (router?.query.docId) {
-      documentQuery.refetch();
-    }
-  }, [router.query]);
+  // TODO: align with BE on an approach to sources and their logos
```

Comment:
I wonder how we link the addition of a new data source, the creation of a new source in the database, the storage of a logo, ... something to talk about I guess :)